src/test/java/org/edumips64/MainCLITest.java

```diff
@@ -0,0 +1,234 @@
+    }
+
+    @Test
+    public void can_show_registers() {
+        OutputStream os = getSystemOut();
+        CommandLine cl = new CommandLine(cli);
+        cl.execute("show", "registers");
+        for (int x = 0; x <= 31; x++) {
+            assertTrue(os.toString().contains("Register "+x+":\t0000000000000000"));
+        }
+    }
+
+    @Test
+    public void can_show_a_register() {
+        OutputStream os = getSystemOut();
+        CommandLine cl = new CommandLine(cli);
+        cl.execute("show", "register", "1");
+        assertTrue(os.toString().contains("0000000000000000"));
+    }
+
+    @Test
+    public void show_register_wont_work_with_alpha_chars() {
+        OutputStream os = getSystemErr();
+        CommandLine cl = new CommandLine(cli);
+        cl.execute("show", "register", "a");
+        assertTrue(os.toString().contains("Invalid value for positional parameter at index 0"));
+    }
+
+    @Test
+    public void prints_usage_message_en() {
+        OutputStream os = getSystemOut();
+        new CommandLine(cli).execute("help");
+        assertEquals(enMainUsageMsg(), os.toString().trim());
+    }
+
+    @Test
+    public void prints_usage_message_it() {
+        OutputStream os = getSystemOut();
+        Locale.setDefault(Locale.ITALIAN);
+        new CommandLine(cli).execute("help");
+        assertEquals(itMainUsageMsg(), os.toString().trim());
+    }
+
+    @Test
+    public void prints_usage_message_show_en() {
+        OutputStream os = getSystemOut();
+        new CommandLine(cli).execute("show", "--help");
+        assertEquals(enShowUsageMsg(), os.toString().trim());
+    }
+
+    @Test
+    public void prints_usage_message_show_it() {
+        OutputStream os = getSystemOut();
+        Locale.setDefault(Locale.ITALIAN);
+        new CommandLine(cli).execute("show", "--help");
+        assertEquals(itShowUsageMsg(), os.toString().trim());
+    }
+
+    @Test
+    public void prints_usage_message_step_en() {
+        OutputStream os = getSystemOut();
+        new CommandLine(cli).execute("step", "--help");
+        assertEquals(enStepUsageMsg(), os.toString().trim());
+    }
+
+    @Test
+    public void prints_usage_message_step_it() {
+        OutputStream os = getSystemOut();
+        Locale.setDefault(Locale.ITALIAN);
+        new CommandLine(cli).execute("step", "--help");
+        assertEquals(itStepUsageMsg(), os.toString().trim());
+    }
+
+    @Test
+    public void can_print_configuration() {
+        new CommandLine(cli).execute("config");
+    }
+
+    private OutputStream getSystemOut() {
+        ByteArrayOutputStream boas = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(boas));
+        return boas;
+    }
+
+    private OutputStream getSystemErr() {
+        ByteArrayOutputStream boas = new ByteArrayOutputStream();
+        System.setErr(new PrintStream(boas));
+        return boas;
+    }
+
+    private String enMainUsageMsg() {
+        return "Usage:  [COMMAND]\n" +
```

Comment:
I'd suggest to not encode the full output of the help here, because as the functionality evolves (adding new commands, changing parameters, etc.) we as authors will need to continuously update these tests to make sure they are up-to-date.

Reply:
Good call, updated to just check sysout is not empty.

Reply:
Could you please push your changes so I can see the updated code?

Reply:
Pushed up most recent changes just now. Still working on some of the other comments you had.